README.md

```diff
@@ -47,6 +47,7 @@ client.quota.callback.static.fetch=1000000
 # after soft limit.
 client.quota.callback.static.storage.soft=10000000
 client.quota.callback.static.storage.hard=20000000
+client.quota.callback.static.storage.percent=95.0
```

Comment:
Don't you need the soft and hard percentage limit as well?

Reply:
Yeah, could be added.
But what would be the quota to return?
Since, honestly, I don't understand this 1.0 being returned. :-)
Where as there is a bound that's Long.MAX ...

Reply:
My understanding of how the limits work right now:
* There are two limits: soft and hard
* After the soft limit is breached, the plugin will start gradually slowing down the producers
* And when the hard limit is reached, it will slow them to 0

This works well with a single disk. But it does not with multiple disks. Because for example in an environment with two disks, it cannot distinguish whether one disk is full and the other empty (=> which would mean a problem because Kafka does not just switch to the other disk automatically) or whether each disk has some share of data and neither of them is full (which would not be a problem).

So we need to introduce something new. My ideas so far were following:
1) We allow users configure the soft/hard limit per disk. For example with something like:
    ```
    client.quota.callback.static.storage.var.lib.kafka.disk-0.soft=10000000000
    client.quota.callback.static.storage.var.lib.kafka.disk-0.hard=20000000000
    client.quota.callback.static.storage.var.lib.kafka.disk-1.soft=40000000000
    client.quota.callback.static.storage.var.lib.kafka.disk-1.hard=50000000000
    ```
    Where the first two would apply to `/var/lib/kafka/disk-0` and the next two to `/var/lib/kafka/disk-1`.
2) Allow configuring the limit as relative number. E.g.:
    ```
    client.quota.callback.static.storage.soft=90
    client.quota.callback.static.storage.hard=95
    ```
    This would be used to evaluate how full the disk is compared to its capacity. It would work exactly the same way as the current quota, but it will be applied on per-log-dir level instead and properly catch the issue with multiple disks. The disadvantage of this is when you have very asymetric storage ... e.g. one disk with 100Gi and one with 10Ti, the single percentage will not work since the difference between 90% and 95% would in one case be 5Gi which is rather small and in the other 500Gi which is rather big. So the first option has definitely more flexibility.

@tombentley @grdryn I think it would be interesting if you can comment how you see it.

Reply:
> And when the hard limit is reached, it will slow them to 0

Yeah, this is where I don't understand that `1.0` being returned when going over hard limit.
Shouldn't it be 0.0? @lulf ?

Reply:
So with 1) approach, I guess we would strip out the path, and find the matching FileStore, and then compare either size or percentage against it?

Is it a config error if you configure two different paths and they end up being in the same FileStore?
Or I guess we can just issue a warning and take min for soft and max for hard limit.

Reply:
@alesj I can't remember exactly, but iirc, it was something in Kafka that didn't like returning 0 or it got stuck with no quota. I think you can reproduce it on a local Kafka with the plugin and producing until reaching the hard limit.

Reply:
@scholzj i've pushed this per-volume changes ...

Reply:
@tombentley OK, feedback applied ... there is just one TODO -- cannot figure out why that regex allows that to pass -- see StaticQuotaCallbackTest ...